runtime/src/serde_snapshot.rs

```diff
@@ -442,51 +451,84 @@ where
+                    if exists {
+                        num_collisions.fetch_add(1, Ordering::Relaxed);
+                    } else {
+                        break (new_append_vec_id, new_append_vec_path);
+                    }
+                };
+                std::fs::rename(append_vec_path, &new_append_vec_path)?;
+
                 reconstruct_single_storage(
                     slot,
-                    append_vec_path,
+                    &new_append_vec_path,
                     storage_entry,
+                    Some(new_append_vec_id),
                     &mut new_slot_storage,
                 )?;
             }
             Ok((*slot, new_slot_storage))
         })
         .collect::<Result<HashMap<Slot, _>, Error>>()?;
+    measure_remap.stop();
 
     // discard any slots with no storage entries
     // this can happen if a non-root slot was serialized
     // but non-root stores should not be included in the snapshot
     storage.retain(|_slot, stores| !stores.is_empty());
+    assert!(
+        !storage.is_empty(),
+        "At least one storage entry must exist from deserializing stream"
+    );
+
+    let next_append_vec_id = next_append_vec_id.load(Ordering::Relaxed);
+    let max_append_vec_id = next_append_vec_id - 1;
+    assert!(
+        max_append_vec_id <= AppendVecId::MAX / 2,
+        "Storage id {} larger than allowed max",
+        max_append_vec_id
+    );
 
+    // Process deserialized data, set necessary fields in self
     accounts_db
         .bank_hashes
         .write()
         .unwrap()
         .insert(snapshot_slot, snapshot_bank_hash_info);
-
-    // Process deserialized data, set necessary fields in self
-    let max_id: usize = *storage
-        .values()
-        .flat_map(HashMap::keys)
-        .max()
-        .expect("At least one storage entry must exist from deserializing stream");
-
-    {
-        accounts_db.storage.0.extend(
-            storage.into_iter().map(|(slot, slot_storage_entry)| {
-                (slot, Arc::new(RwLock::new(slot_storage_entry)))
-            }),
-        );
-    }
-
-    if max_id > AppendVecId::MAX / 2 {
-        panic!("Storage id {} larger than allowed max", max_id);
-    }
-
-    accounts_db.next_id.store(max_id + 1, Ordering::Relaxed);
+    accounts_db.storage.0.extend(
+        storage
+            .into_iter()
+            .map(|(slot, slot_storage_entry)| (slot, Arc::new(RwLock::new(slot_storage_entry)))),
+    );
+    accounts_db
+        .next_id
+        .store(next_append_vec_id, Ordering::Relaxed);
     accounts_db
         .write_version
         .fetch_add(snapshot_version, Ordering::Relaxed);
     accounts_db.generate_index(limit_load_slot_count_from_snapshot, verify_index);
+
+    datapoint_debug!(
+        "reconstruct_accountsdb_from_fields()",
+        ("remap-time-us", measure_remap.as_us(), i64),
+        (
+            "remap-collisions",
+            num_collisions.load(Ordering::Relaxed),
+            i64
+        ),
+    );
```

Comment:
I figured these datapoints would be helpful too. Knowing how long the remapping takes may later indicate if/when this needs to be sped up. Also, I think the number of collisions will be an interesting metric.

Reply:
`datapoint_debug!` won't show up by default in most reporting, should use at least `datapoint_info!`

Reply:
Oh, that's good to know. What/where is the "reporting" that you're referring to? I'd like to be able to see these datapoints on the grafana/metrics website (not the local console logs); that one needs `_info!` or higher?

Reply:
yup, info! or higher